Static site

keynmol opened the pull request:
Closes #61 

**Published site: http://laminar-docusaurus.surge.sh/** (will change as I progress with the PR)

* [x] uncomment build settings (somehow versions interfere)
* [x] remove unused images
* [ ] more content?
* [ ] setup circleto publish on release

lolgab commented on .gitignore:
```diff
@@ -4,3 +4,12 @@ target
+
+project/.bloop
+
+website/node_modules
```
Trailing newline missing here.

lolgab commented on build.sbt:
```diff
@@ -1,30 +1,55 @@
 
-useYarn := true
+lazy val docs = project       // new documentation project
+  .in(file("myproject-docs")) // important: it must not be docs/
```
If it must not be `docs` you could rename the project `lazy val` to something else too.
Like `laminar-docs` or `laminarDocs`

keynmol replied:
This name actually doesn't matter, that's where mdoc will put compiled documentation:

https://scalameta.org/mdoc/docs/installation.html#sbt

I'll rename the folder to `laminar-docs`, but it's only a temporary one anyways.

lolgab commented on build.sbt:
```diff
@@ -1,30 +1,55 @@
+  )
 
-requireJsDomEnv in Test := true
+lazy val laminar = project.in(file("."))
```
Does it make sense to move laminar code to a `laminar` subfolder?

keynmol replied:
I'll defer to @raquo's preference on this one - technically there's no need to do this as it's still a single module and a single deliverable, and I can imagine that moving all the sources in this PR would make it more daunting  :)

It would make more sense if Airstream wasn't a separate library and was just a module here, which would probably make releasing cascade simpler, and means the docs could include both airstream and laminar examples

raquo replied:
Hmm for now let's keep Laminar in `.`, and I'll take a look at this again when reviewing all this at the end.

Moving Airstream into this git repo sounds compelling, but I'm not sure how to deal with git version tags in that case since Laminar version doesn't always match Airstream version, and git tags are currently the primary navigation mechanism for looking at the code or docs for a particular version.

lolgab commented on docs/Documentation.md:
```diff
@@ -1,54 +1,63 @@
-  * [How Are Mount Events Propagated?](#how-are-mount-events-propagated)
-* [URL Routing](#url-routing)
-* [Special Cases](#special-cases)
+- [Introduction](#introduction)
+    - [First, see Why Laminar and [Quick Start](../README.md#quick-start).](#first-see-why-laminar-and-quick-start)
+    - [Documentation](#documentation)
+- [Imports](#imports)
+- [Tags & Elements](#tags--elements)
+- [Modifiers](#modifiers)
+  - [Nesting and Children](#nesting-and-children)
+  - [Empty Nodes](#empty-nodes)
+  - [Manual Application](#manual-application)
+  - [`inContext`](#incontext)
+  - [Reusing Elements](#reusing-elements)
+  - [Missing Keys](#missing-keys)
+  - [Modifiers FAQ](#modifiers-faq)
+- [Rendering](#rendering)
+- [Reactive Data](#reactive-data)
+  - [Attributes and Properties](#attributes-and-properties)
```
I think the Markdown formatting changes should go, in case you really want them, to a separate PR.

keynmol replied:
o_O

I actually didn't do this, it must be something in my markdown VS code setup.

All I wanted was just to remove the header, I'll revert.

lolgab reviewed (changes requested):
Sorry, I started reviewing before realising this is still a draft PR!

keynmol commented:
@raquo is Circle going to pick up a PR from fork if I move this into a reviewable state?

raquo commented:
@keynmol I updated circle CI config, and checked that it triggers on every commit in every branch now. I think if you rebase your PR on latest master, it will start being tested. I hope that github checks integration will also start working then.

raquo commented:
@keynmol Ok I just tested with https://github.com/raquo/Laminar/pull/64, circle CI integration works on latest master. You'll need to log in to circle CI with your github account to view the build logs though (no special permissions other than that required).

keynmol commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+{
```
TODO: delete this :-/

keynmol commented:
@lolgab I think I've addressed your comments - please re-review if that's not the case

@raquo 

So the current state is:

- ✅ mdoc is wired in correctly (and I fixed bugs that I myself introduced in mdoc :D)
- ✅ Site is successfully built on master
- ✅ locally I can [push the whole site](http://laminar-docusaurus.surge.sh/) and it is what I expect

Now there's a few outstanding questions:

- Content? 

   I think it's up to you what you want to see on the main page, in the docs, in the examples (I can help write the examples, just don't know what sort of topics you want to showcase). You can take a look at the existing ones - apart from 1 html hack to make them side by side it's very simple. In fact a whole TodoApp from laminar-examples can just be fit into the snippet - wouldn't that be cool :D

- Logo and color scheme?
  
  Right now the answer is "spiral from Inkscape" and "greenish" :) 

- Releasing?

   I'm not seeing release specific settings in circle - which suggests you're publishing Laminar manually. 

   As long as everything is set up locally like CI, you can [publish locally](https://scalameta.org/mdoc/docs/docusaurus.html#publish-to-github-pages-locally) or we can attempt to configure it to [publish from Circle CI](https://scalameta.org/mdoc/docs/docusaurus.html#publish-to-github-pages-from-ci) - your choice

raquo commented on build.sbt:
```diff
@@ -1,27 +1,85 @@
+lazy val jsdocs = project
+  .settings(
+   libraryDependencies += "org.scala-js" %%% "scalajs-dom" % "1.1.0",
+   scalaVersion := Scala212,
```
Is anything preventing us from using 2.13?

keynmol replied:
Hm, I think it's an oversight in mdoc - js modifier is only published for 2.12. I'm going to [rectify that](https://github.com/scalameta/mdoc/pull/385)

Until that's published, I can force 2.12 during the site building, but keep laminar's own default Scala version at 2.13 (not that it matters too much, given it's cross-publishing)

raquo commented:
@keynmol Thanks for all your work! I can take care of initial content and design. Currently I'm releasing Laminar from my computer, running `sbt release` and then `sbt sonatypeRelease`. Didn't get to setting up anything fancier. What are the commands for building and publishing the docs?

```
sbt docs/docusaurusCreateSite
cd docs
npm run docusaurus-build
npm run docusaurus-publish
``` 

Is that right?

keynmol commented:
After building the site from sbt, `cd website && yarn install && USE_SSH=true yarn publish-gh-pages`

That's what mdoc site says - may be its worth publishing a test site out of this branch first?

raquo commented:
The force-push just now was only to remove my test commit, no other history changed.
